### PR TITLE
Nuke: Fix workfile template builder so representations get loaded next to each other

### PR DESCRIPTION
**Pull request by fabiaserra**
## Changelog Description
Refactor when the cleanup of the placeholder happens for the cases where multiple representations are loaded by a single placeholder.

The existing code didn't take into account the case where a template placeholder can load multiple representations so it was trying to do the cleanup of the placeholder node and the re-arrangement of the imported nodes too early. I assume this was designed only for the cases where a single representation can load multiple nodes.

Screenshot 2023-05-23 170233

## Additional info
Removed the `failed` argument from `cleanup_placeholder` as it wasn't being used, we can add it if it's ever needed.

## Testing notes:
1. Create a Nuke workfile template with a placeholder that will load multiple representations (i.e. Using Load Clip loader for example)
2. Save workfile template at a place where project Nuke workfile template builder settings looks for
3. Create workfile from template

### CODE REVIEW

**Review by Minkiu — APPROVED** [review]
Haven't tested, code looks  good. I trust you replaced all instances of the changed methods hehe

**Comment by fabiaserra** [comment]
> Haven't tested, code looks good. I trust you replaced all instances of the changed methods hehe otherwise might be worth keeping the arguments as optional and issue a warning of deprecation.

Yeah good call, apparently someone cared even less than me for backwards compatibility and changed the whole function name here https://github.com/ynput/OpenPype/pull/4854 haha so this comment no longer really applies...

I have now merged `develop` over and resolved the merge conflicts and I have kept the existing function signature with the redundant `failed` arg that no one is using...

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 1]
```diff
@@ -1549,24 +1559,23 @@ def populate_load_placeholder(self, placeholder, ignore_repre_ids=None):
...
             except Exception:
-                failed = True
                 self.load_failed(placeholder, representation)
 
             else:
-                failed = False
                 self.load_succeed(placeholder, container)
-            self.post_placeholder_process(placeholder, failed)
+        # Run post placeholder process after load of all representations
+        self.post_placeholder_process(placeholder, failed)
```
[the remark this answers is not in the document]
Comment (fabiaserra) [reply openpype/pipeline/workfile/workfile_template_builder.py:1578]: and yeah, this was going to be my next point, `failed` is undefined here but so it is on the next line that checks `if failed`

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 2]
```diff
@@ -1549,24 +1559,23 @@ def populate_load_placeholder(self, placeholder, ignore_repre_ids=None):
...
 
         if failed:
```
Comment (fabiaserra) [inline openpype/pipeline/workfile/workfile_template_builder.py:1580]: here `failed` doesn't necessarily exist on this scope. @tokejepsen tagging you as you were the one who made that commit, what were you expecting this to be like? if the first representation fails but the next one works, the value of `failed` gets overridden

Comment (fabiaserra) [reply openpype/pipeline/workfile/workfile_template_builder.py:1583]: I have now pushed here https://github.com/ynput/OpenPype/pull/5061/commits/ac7e53a44a038df8c31ad1fbd4d49c8c90c3a95b a change so `failed` is defined on the right scope and it only gets overridden to `True` if it fails for any of the load calls

**Review by jakubjezek001 — APPROVED** [review]
Tested and works well! Thanks @fabiaserra. Since Maya changes are only related to docstrings changes. I am assuming we can merge it!

File: openpype/pipeline/workfile/workfile_template_builder.py (hunk 2 above)
Comment (jakubjezek001) [reply openpype/pipeline/workfile/workfile_template_builder.py:1596]: It seems to be resolved @fabiaserra, right?

**Pull request merged**